compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/codegen/JvmTypeGen.java

```diff
@@ -900,9 +901,13 @@ private void loadUserDefinedType(MethodVisitor mv, BType bType) {
         String typeOwner = JvmCodeGenUtil.getPackageName(packageID) + MODULE_INIT_CLASS_NAME;
         String fieldName = getTypeFieldName(toNameString(typeToLoad));
 
+        HashMap<String, String> encodedVsInitialIds = new HashMap<>();
+        JvmDesugarPhase.encodePackageIdentifiers(packageID, encodedVsInitialIds);
```

Comment:
Encode, compare and then encode again looks really weird to me.
@warunalakshitha is this normal thing to do? Or else would it be better to extract a function that manually compare  between encoded and unrecorded one. Or is this a bug that `packageId` is not encoded here?

Reply:
Yes. This seems not valid. I think we can just encode and compare. IF not keep the bother encoded packageId as a class variable.